crates/sled/src/tree.rs

```diff
@@ -210,21 +210,26 @@ impl Tree {
     /// let t = sled::Db::start(config).unwrap();
     ///
     /// // unique creation
-    /// assert_eq!(t.cas(&[1], None, Some(vec![1])), Ok(Ok(())));
+    /// assert_eq!(t.cas(&[1], None as Option<&[u8]>, Some("1")), Ok(Ok(())));
     ///
     /// // conditional modification
-    /// assert_eq!(t.cas(&[1], Some(&*vec![1]), Some(vec![2])), Ok(Ok(())));
+    /// assert_eq!(t.cas(&[1], Some("1"), Some("2")), Ok(Ok(())));
     ///
     /// // conditional deletion
-    /// assert_eq!(t.cas(&[1], Some(&[2]), None), Ok(Ok(())));
+    /// assert_eq!(t.cas(&[1], Some("2"), None as Option<&[u8]>), Ok(Ok(())));
     /// assert_eq!(t.get(&[1]), Ok(None));
     /// ```
-    pub fn cas<K: AsRef<[u8]>>(
+    pub fn cas<K, OV, NV>(
         &self,
         key: K,
-        old: Option<&[u8]>,
-        new: Option<Value>,
-    ) -> Result<std::result::Result<(), Option<IVec>>> {
+        old: Option<OV>,
+        new: Option<NV>,
+    ) -> Result<std::result::Result<(), Option<IVec>>>
+    where
+        K: AsRef<[u8]>,
+        OV: AsRef<[u8]>,
```

Comment:
Don't you remenber that it was not possible the last time I tried? 😄
I do not think that the Rust language team changed that inference behavior, sadly...

https://github.com/spacejam/sled/pull/422#discussion_r233732645

Reply:
Ok, we will keep these lacks of inference. I prefer that, it is more generic!

Reply:
cas is already sort of a rare codepath and today I'm preferring genericism